semantics.md

```diff
@@ -76,24 +76,30 @@ String literals exist, and use curly braces for string interpolation. They may a
 
 Numbers, and the boolean constants true and false exist as normal. Numbers are represented internally as floating points.
 
-## Commands
+## Minecraft Interface - Triggers and Commands
 
-A grant command expects a string naming an advancement, and is translated into a command to grant that advancement.
+### Triggers
 
-A revoke command expects a string naming an advancement, and is translated into a command to revoke that advancement.
+Triggers run actions when a certain event happens within Minecraft. Internally, these are implemented using [built-in function tags](https://minecraft.fandom.com/wiki/Function_(Java_Edition)#Tags) for `load` and `tick`, and hidden advancement triggers for all other triggers. For the latter, the hidden advancement is revoked from the player right after it triggers, so that the same event can be triggered multiple times.
 
-An execute command expects a string naming a function, and is translated into a command to run that (Minecraft) function.
+Of these, `load` and `tick` are special and cannot be combined together. `load` triggers whenever the datapack loads, and `tick` triggers on every Minecraft game tick. Other triggers are documented on the [Minecraft Wiki](https://minecraft.fandom.com/wiki/Advancement/JSON_format) and can be combined together using an or statement (`||`) - so the on-expression's commands will run when any specified trigger fires.
 
-A literal command expects a string, and is inserted verbatim. No type checks happen.
+The typechecked triggers (currently `consume_item` and `inventory_changed`) take item specifications - we'd like to typecheck that the item or tag are valid items, but we can't do that without listing all the items and tags from Minecraft.
 
-## Triggers
+The free-form trigger takes a Minecraft trigger name - we'd like to typecheck that it's a valid name, but we can't do that without listing all of the triggers from Minecraft. This also doesn't allow parameters on the trigger.
 
-The load trigger is special, as it can't be combined at all. The trigger will fire when the datapack is reloaded or the server is started. (This is implemented using function tags.)
+`on` blocks cannot be nested.
 
-The tick trigger is also special, as it can't be combined at all. The trigger will fire every single tick. (This is implemented using function tags.)
+### Commands
 
-Triggers can be combined together using an or - so the on-expression's commands will run when any trigger fires.
+Commands are triggered in Minecraft through the `run` statement, which allows for the following subcommands:
```

Comment:
Why are we adding this? We just need to modify raw commands to accept lists of strings. We could also make `grant`, `revoke`, and `execute` produce strings - just for consistency.